File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/Assault.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.annotations.ApiModel;
import lombok.*;

import java.time.LocalDate;

@ApiModel(description = "Assault")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString
@Data
public class Assault {

    private String offenderNo;
    private String type;
    private LocalDate dateCreated;
    private boolean serious;

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/Assault.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import io.swagger.annotations.ApiModel
import java.time.LocalDate

@ApiModel(description = "Assault")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class Assault(
  private val offenderNo: String? = null,
  private val type: String? = null,
  private val dateCreated: LocalDate? = null,
  private val serious: Boolean = false,
)
